### PR TITLE
Supported Currency and added additional tests for Vector, Stack and Properties.

### PR DESCRIPTION
**Pull request by inder123**

### CODE REVIEW

**Comment by inder123** [comment]
@swankjesse  review please?

File: gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java [hunk 1]
```diff
@@ -576,6 +577,18 @@ public void write(JsonWriter out, UUID value) throws IOException {
 
   public static final TypeAdapterFactory UUID_FACTORY = newFactory(UUID.class, UUID);
 
+  public static final TypeAdapter<Currency> CURRENCY = new TypeAdapter<Currency>() {
```
Comment (swankjesse) [inline gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java]: Is anyone asking for this?!

**Comment by swankjesse** [comment]
LGTM, though I'm quite surprised that you're adding such a bogus type.

Also, please squash.

**Comment by inder123** [comment]
No one asking for it, but I noticed that the default serialization was not correct.

**Comment by swankjesse** [comment]
Got it. Slightly scary, this change is not interoperable to anyone (accidentally) relying on the current broken behavior. I like Moshi's behavior here, which is to throw an exception! `Platform class java.util.Vector annotated [] requires explicit JsonAdapter to be registered`

**Comment by inder123** [comment]
@swankjesse  I only added support for Currency, not vector.
Yes, it breaks with the current behavior, but I think this is what any normal user will expect. In the past, we have added default adapters for classes from java.util.* packages over time, technically breaking the serialization semantics.

**Comment by inder123** [comment]
merging based on earlier LGTM

**Pull request merged**